src/gatsby-ssr.js

```diff
@@ -24,12 +24,33 @@ exports.onRenderBody = ({ setHeadComponents }, pluginOptions) => {
     }}();
   `;
 
+    const delayedLoader = `
+      window.segmentSnippetLoaded = false;
+      window.segmentSnippetLoading = false;
+
+      window.segmentSnippetLoader = (callback) => {
+        if (!window.segmentSnippetLoaded && !window.segmentSnippetLoading) {
+          window.segmentSnippetLoading = true;
+          setTimeout(() => {
+            ${snippet}
+            window.segmentSnippetLoading = false;
+            window.segmentSnippetLoaded = true;
+            if(callback) {callback()}
+          }, 1000);
```

Comment:
@davej, any particular reason why you chose 1s?

Reply:
@Kilian

I actually wait 3 seconds and then I wrap the function in `requestIdleCallback` too.  The audience for that blog post is not super-tech so the author decided to simplify the code example. Using this script will not capture bounce-rate so that's a trade-off and it's best to capture bounce rate in a different way (perhaps server logs). YMMV on the 3 seconds though, see what works for you. You probably don't want to do it immediately on scroll because it might make the scrolling janky.

One other note on usage, you may have picked up on this already. It's best to include the Segment snippet normally and then call the `.load()` method in the delayed function. If you do this then you won't drop/error any events that happen before the scroll.

Here's the code that I'm using:
```js
window.waitBeforeCalling = (fn, ms = 3000) => (...args) => {
  window.addEventListener(
    "scroll",
    () =>
      setTimeout(
        () =>
          "requestIdleCallback" in window
            ? requestIdleCallback(() => fn(...args))
            : fn(...args),
        ms
      ),
    { once: true }
  );
};
```

And usage:

```js
!function(){var analytics=window.analytics=window.analytics||[];if(!analytics.initialize)if(analytics.invoked)window.console&&console.error&&console.error("Segment snippet included twice.");else{analytics.invoked=!0;analytics.methods=["trackSubmit","trackClick","trackLink","trackForm","pageview","identify","reset","group","track","ready","alias","debug","page","once","off","on"];analytics.factory=function(t){return function(){var e=Array.prototype.slice.call(arguments);e.unshift(t);analytics.push(e);return analytics}};for(var t=0;t<analytics.methods.length;t++){var e=analytics.methods[t];analytics[e]=analytics.factory(e)}analytics.load=function(t,e){var n=document.createElement("script");n.type="text/javascript";n.async=!0;n.src="https://cdn.segment.com/analytics.js/v1/"+t+"/analytics.min.js";var a=document.getElementsByTagName("script")[0];a.parentNode.insertBefore(n,a);analytics._loadOptions=e};analytics.SNIPPET_VERSION="4.1.0";
  waitBeforeCalling(() => {
    analytics.load('INSERT_KEY_HERE');
    analytics.page()
  }, 500)();
}}();
```

Reply:
Oh looks like I actually changed the wait time to 500ms in the code example above. I can't remember doing that. My advice is to test out a few different configurations and see what works best.

Reply:
check, so two changes coming up:

- switch to only delay the .load call
- (optionally?) let people choose their own delay time and default to something reasonable (1s is as good as any?)

Reply:
I would also consider wrapping in `requestIdleCallback` where available.

Reply:
Updated the script to include the above:

- there is now an optional delayLoadTime setting, which defaults to 1s
- we're only delaying the .load() call
- the function is called in a requestIdleCallback when available.